### PR TITLE
Support New OTP2 Leg Step Types

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Adds front-end support for https://github.com/opentripplanner/OpenTripPlanner/pull/5285

Icons are quite rudimentary, but can be improved in the future

### CODE REVIEW

File: packages/itinerary-body/i18n/fr.yml [hunk 1]
```diff
@@ -49,6 +52,7 @@ otpUi:
       southeast: le sud-est
       southwest: le sud-ouest
       west: l'ouest
+    stepStation: "{step} TODO {street}"
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/i18n/fr.yml:55]:
```suggestion
    stepStation: "{step} par {street}"
```

File: packages/itinerary-body/src/__mocks__/itineraries/leg-fare-products.json [hunk 2]
```diff
@@ -54,6 +54,21 @@
         }
       },
       "steps": [
+        {
+          "relativeDirection": "ENTER_STATION",
+          "stayOn": false,
+          "streetName": "fake station name"
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/__mocks__/itineraries/leg-fare-products.json:60]:
```suggestion
          "streetName": "fake station entrance name"
```

File: packages/itinerary-body/src/__mocks__/itineraries/leg-fare-products.json [hunk 3]
```diff
@@ -54,6 +54,21 @@
...
+        },
+        {
+          "relativeDirection": "FOLLOW_SIGNS",
+          "stayOn": false,
+          "streetName": "fake station exit"
+        },
+        {
+          "relativeDirection": "EXIT_STATION",
+          "stayOn": false,
+          "streetName": "fake station name"
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/__mocks__/itineraries/leg-fare-products.json:70]:
```suggestion
          "streetName": "fake station entrance name"
```

**Review by binh-dam-ibigroup — APPROVED** [review]
Please see the content suggestions, looks good otherwise.

**Review by amy-corson-ibigroup — APPROVED** [review]

**Pull request merged**